### DISCUSSION TITLE
400 error if app uses /signin endpoint

### DISCUSSION THREAD

**Discussion by treeder** [community/community#5204] [Codespaces]
I have a web app that uses `/signin` to let user's sign in, when I run this on Codespaces and try to view the page it returns a 400 error with this as the contents:

```
{"type":"https://tools.ietf.org/html/rfc7231#section-6.5.1","title":"Bad Request","status":400,"traceId":"|f82d6b3c-45fdf8c953526095."}
```

Thinking this was my app, I tried various things including changing it to `/signin2` (which also doesn't work). Then I tried putting that same page under something totally different like `/abcxyz` and then it works. 

So it looks like GitHub is getting in the middle of that particular endpoint prefix. Which will probably break a lot of apps since that's a pretty common endpoint.

**Comment by treeder** [comment]
Nobody else getting this?

**Reply by thalida** [reply]
Also seeing this with my app -- I switched to using "login"/"logout" as a workaround. I need `https` support so I can't got back to using the localhost which doesn't have this problem.

**Reply by rajapars** [reply]
I get this error also. Thanks for your info.

**Comment by tanmayeekamath** [comment]
Hey @treeder, thanks for your feedback. 
This is currently a known limitation and we have a workitem to fix this on our side. Up until then the workaround would be to use a prefix other than `/signin`,` /auth`, `/authenticate-codespace/*` as you pointed out. 
We will also update our docs FAQ section to reflect this until a fix is deployed.

**Reply by DiesIrae** [reply]
It's not customizable in KeystoneJS, so we can't use Codespaces to develop KeystoneJS projects.

https://github.com/keystonejs/keystone/discussions/8394

**Comment by ideaoforder** [comment]
So, they just gave up on this? Seems like this would affect A LOT of codespaces users...

**Reply by RafeSl** [reply]
We had a meeting with GH a number of months ago and ... short version... "not an easy fix"
My PERSONAL guess is that there is some hardware that is using /signin and that it can't be easily intercepted...

**Comment by rajapars** [comment]
I also get this error status 400 when access /signin endpoint.
I shall use another endpoint, but this impact to development because will need to update it later before commit to git. Hmm...
